### PR TITLE
Prevent actor import dupe relations, refs #13148

### PR DESCRIPTION
**Pull request by mcantelon**
Added checks for pre-existing relations during authority record
relation import.

### CODE REVIEW

File: lib/task/import/csvAuthorityRecordImportTask.class.php [hunk 1]
```diff
@@ -535,29 +535,53 @@ public function execute($arguments = array(), $options = array())
...
-                if ($self->rowStatusVars['startDate'])
-                {
-                  $relation->startDate = $self->rowStatusVars['startDate'];
-                }
-                if ($self->rowStatusVars['endDate'])
+                // Check for pre-existing relationship before adding
+                $sql = "SELECT id FROM relation \r
+                  WHERE subject_id=:subject_id \r
+                  AND object_id=:object_id \r
+                  AND type_id = :type_id";
```
Comment (sbreker) [inline lib/task/import/csvAuthorityRecordImportTask.class.php:542]: Nitpick: Extra spaces around '=' compared to prev lines.

File: lib/task/import/csvAuthorityRecordImportTask.class.php [hunk 2]
```diff
@@ -535,29 +535,53 @@ public function execute($arguments = array(), $options = array())
...
+                  ':subject_id' => $sourceActorId,
+                  ':object_id' => $targetActorId,
+                  ':type_id' => $relationTypeId
+                );
+
+                $paramsVariant = array(
+                  ':subject_id' => $targetActorId,
+                  ':object_id' => $sourceActorId,
+                  ':type_id' => $relationTypeId
+                );
```
Comment (sbreker) [inline lib/task/import/csvAuthorityRecordImportTask.class.php:554]: Does it matter that this relation is the opposite of the one being created? Or is this the source of the duped relations?

File: lib/task/import/csvAuthorityRecordImportTask.class.php (hunk 1 above)
Comment (mcantelon) [reply lib/task/import/csvAuthorityRecordImportTask.class.php:542]: Good spotting... I've added the spaces to all of them as it makes it a bit easier to read.

File: lib/task/import/csvAuthorityRecordImportTask.class.php (hunk 2 above)
Comment (mcantelon) [reply lib/task/import/csvAuthorityRecordImportTask.class.php:554]: I think a relationship can be specified in two ways, so I'm checking for both.

**Comment by mcantelon** [comment]
I've updated the PR... thanks @sbreker !

**Review by sbreker — APPROVED** [review]

**Pull request merged**